addons/website_slides/static/src/js/slides_slide_archive.js

```diff
@@ -46,8 +52,11 @@ var SlideArchiveDialog = Dialog.extend({
             params: {
                 slide_id: this.slideId
             },
-        }).then(function () {
-            self.$slideTarget.closest('.o_wslides_slides_list_slide').remove();
+        }).then(function (data) {
+            if (!data.error){
+                self.$slideTarget.closest('.o_wslides_slides_list_slide').remove();
```

Comment:
As said in your controller, you should be able to play with the existing return value (True/False) to avoid breaking stable.

For the second part you will not have the "category_id" thing, is there any way to find out using the HTML instead? Or some other existing data?